DXMainClient/DXGUI/Multiplayer/ChatListBox.cs

```diff
@@ -17,9 +19,50 @@ public class ChatListBox : XNAListBox, IMessageView
         public ChatListBox(WindowManager windowManager) : base(windowManager)
         {
             DoubleLeftClick += ChatListBox_DoubleLeftClick;
+            MiddleClick += ChatListBox_MiddleClick;
         }
-
+        
         private void ChatListBox_DoubleLeftClick(object sender, EventArgs e)
+        {
+            if (SelectedIndex < 0 || SelectedIndex >= Items.Count)
+                return;
+
+            var link = Items[SelectedIndex].Text?.GetLink();
+            if (link == null)
+                return;
+
+            bool regExpResult = new Regex(ClientConfiguration.Instance.AlwaysTrustedLinksRegExp).Match(link).Success ||
+                new Regex(ClientConfiguration.Instance.TrustedLinksRegExp).Match(link).Success;
```

Comment:
new Regex() is time consuming. Consider reuse the regexp

Reply:
Done by 734a928c5047c6c2bfab83d4a67bd960cb71c9e7.